lib/spack/spack/util/environment.py

```diff
@@ -78,9 +83,17 @@ def path_put_first(var_name, directories):
 
 def dump_environment(path):
     """Dump the current environment out to a file."""
+    hidden_keys = set(['PS1', 'PWD', 'OLDPWD', 'TERM_SESSION_ID'])
+    keys = sorted(os.environ)
+
     with open(path, 'w') as env_file:
-        for key, val in sorted(os.environ.items()):
-            env_file.write('export %s="%s"\n' % (key, val))
+        for key in keys:
+            if key.endswith('%%'):
```

Comment:
Regarding https://github.com/spack/spack/pull/11133#issuecomment-481696756 and https://github.com/spack/spack/pull/11133#issuecomment-481942483: 

> This (I believe) duplicates functionality provided by #8476.

There is overlap (e.g. use of `shlex_quote`) but this adds some functionality (e.g. this line, which avoids adding bash functions).

The PRs can likely be merged in either order, but whichever one goes last will have a conflict.

Reply:
@scheibelp I think this branch is ready for review/merge. Is there anything that needs to be added or changed?